src/plots/declarative.py

```diff
@@ -361,21 +451,110 @@ def _set_yscale(self, ax, yscale):
         """
         ax.set_yscale(yscale)
 
+    def _sharex(self, ax):
+        """
+        If sharex axis is True, will find where to hide xticklabels.
+        """
+        if ax.is_last_row():
+            pass
+        else:
+            plt.setp(ax.get_xticklabels(), visible=False)
+
+    def _sharey(self, ax):
+        """
+        If sharey axis is True, will find where to hide yticklabels.
+        """
+        if ax.is_first_col():
+            pass
+        else:
+            plt.setp(ax.get_yticklabels(), visible=False)
+
+    def _add_map_features(self, ax, map_features):
+        """
+        Factory to add map features.
+        """
+        feature_dict = {
+            'coastlines': self._add_coastlines,
+            'borders': self._add_borders,
+            'states': self._add_states,
+            'lakes': self._add_lakes,
+            'rivers': self._add_rivers,
+            'land': self._add_land,
+            'ocean': self._add_ocean
+        }
+
+        for feat in map_features:
+            try:
+                feature_dict[feat](ax)
+            except KeyError:
+                raise TypeError(f'{feat} is not a valid map feature.' +
+                                'Current map features supported are:\n' +
+                                f'{" | ".join(feature_dict.keys())}"')
+
+    def _add_coastlines(self, ax):
```

Comment:
Can we make this into a factory such that you just pass the add_feature(cfeature.COASTLINE)?

Reply:
Meaning in your dict above, put `cfeature.COASTLINE` and then just one `def _add_feature`

Reply:
I'll do you one up: 
```
feature_dict = {
    'coastline': cfeature.COASTLINE,
    'borders': cfeature.BORDERS,
    'states': cfeature.STATES,
    'lakes': cfeature.LAKES,
    'rivers': cfeature.RIVERS,
    'land': cfeature.LAND,
    'ocean': cfeature.OCEAN
}

for feat in map_features:
    try:
        ax.add_feature(feature_dict[feat])
    except:
        ....
```